osu.Game/Rulesets/Ruleset.cs

```diff
@@ -380,5 +380,22 @@ public IEnumerable<(HitResult result, LocalisableString displayName)> GetHitResu
         /// Can be overridden to add a ruleset-specific section to the editor beatmap setup screen.
         /// </summary>
         public virtual RulesetSetupSection? CreateEditorSetupSection() => null;
+
+        /// <summary>
+        /// <para>
+        /// The Ruleset Online ID that maps need to have to be able to get
+        /// converted from. Defaults to 0 (osu! maps), but can be changed to
+        /// other IDs to not use osu! maps, but instead only use taiko (1),
+        /// catch (2) or mania (3) maps as source maps.
+        /// </para>
+        /// <para>
+        /// Furthermore the player needs to allow converted maps in the settings
+        /// in order for converted maps to show, as well as the ruleset map
+        /// convertor returned by
+        /// <see cref="Ruleset.CreateBeatmapConverter(IBeatmap)"/>
+        /// needing to be able to convert them.
+        /// </para>
+        /// </summary>
+        public virtual int ConversionSourceRulesetID => 0;
```

Comment:
This is not happening. We don't expose these legacy `RulesetID`s anywhere else.

If anything, it should be using `ShortName`. But as already said in review, we would want to think this through better. This only allows one choice for the "source", and means that the majority of beatmaps will.. no longer convert? That's against the whole premise of how osu! works. So if anything, this should be "additional rulesets to allow as a conversion source".

Reply:
yes that's correct, although it's only for rulesets that choose to not support converting from regular osu maps and instead only convert from other rulesets exclusively.

I think specifying them as additional rulesets would be more useful as well and am going to try making a well performing, more generic, solution later.

I made a local 2-player taiko ruleset based on the taiko ruleset, where I wanted to show taiko maps instead of osu maps, but this is mostly more a hack than what a real implementation should look like.